src/ui/components/filters/filteroptionscomponent.js

```diff
@@ -110,17 +110,36 @@ class FilterOptionsConfig {
         config.previousOptions = [];
       }
     }
-    let selectedOptions = config.previousOptions || [];
-    this.options = this.setDefaultSelectedValues(this.options, selectedOptions);
+    const previousOptions = config.previousOptions || [];
+    this.options = this.setDefaultSelectedValues(this.options, previousOptions);
   }
 
-  setDefaultSelectedValues (options, selectedOptions) {
-    return options.map(o => ({
-      ...o,
-      selected: selectedOptions.length
-        ? selectedOptions.includes(o.label)
-        : o.selected
-    }));
+  /**
+   * Sets selected options on load based on options stored in persistent storage,
+   * and if persistent storage is empty then the appliedOnLoad flag.
+   * @param {Array<Object>} options
+   * @param {Array<string>} previousOptions
+   * @returns {Array<Object>}
+   */
+  setDefaultSelectedValues (options, previousOptions) {
+    if (previousOptions.length) {
+      return options.map(o => ({
+        ...o,
+        selected: previousOptions.length
```

Comment:
This check seems redundant.

Reply:
This check is here so that if there are any options in persistentStorage, we use those. Otherwise we use appliedOnLoad

Reply:
Ah okay you're right sorry about that. This isn't really an excuse but Github made it look like you were talking about the first check
![image](https://user-images.githubusercontent.com/23005393/82685336-08485280-9c22-11ea-96d6-488f9022b759.png)